packages/miniflare/src/workers/kv/assets.worker.ts

```diff
@@ -0,0 +1,34 @@
+import { SharedBindings } from "miniflare:shared";
+import { decodeAssetsKey, KVParams } from "./constants";
+
+interface Env {
+	[SharedBindings.MAYBE_SERVICE_BLOBS]: Fetcher;
+}
+
+export default <ExportedHandler<Env>>{
+	async fetch(request, env) {
+		// Only permit reads
+		if (request.method !== "GET") {
+			const message = `Cannot ${request.method.toLowerCase()}() with Workers Assets namespace`;
+			return new Response(message, { status: 405, statusText: message });
+		}
+
+		// Decode key
+		const url = new URL(request.url);
+		let key = url.pathname.substring(1); // Strip leading "/"
+		if (url.searchParams.get(KVParams.URL_ENCODED)?.toLowerCase() === "true") {
+			key = decodeURIComponent(key);
+		}
```

Comment:
Is this possible in the case of assets (compared to sites, etc)

Reply:
tested, and yes, we do need to decode the key
<img width="450" alt="Screenshot 2024-08-13 at 19 28 21" src="https://github.com/user-attachments/assets/99f121cf-8f8f-4c12-88a8-31376f3bdf91">